File: 본문/cpp/Code_4_04_1.cpp
#include <iostream>
using namespace std;

int N;
bool prime[100000009];

int main() {
	// 입력 → 배열 초기화
	cin >> N;
	for (int i = 2; i <= N; i++) prime[i] = true;

	// 에라토스테네스의 체
	for (int i = 2; i * i <= N; i++) {
		if (prime[i] == true) {
			for (int x = 2 * i; x <= N; x += i) prime[x] = false;
		}
	}

	// N 이하의 소수를 오름차순으로 출력
	for (int i = 2; i <= N; i++) {
		if (prime[i] == true) cout << i << endl;
	}
	return 0;
}

File: 본문/python/Code_4_04_1.py
# 입력 → 배열 prime 초기화
N = int(input())
prime = [ True ] * (N + 1)

# 에라토스테네스의 체
LIMIT = int(N ** 0.5)
for i in range(2, LIMIT + 1):
	if prime[i] == True:
		# x = 2i, 3i, 4i, ...로 N 이하의 범위에서 반복
		for j in range(2 * i, N + 1, i):
			prime[j] = False

# N 이하의 소수를 오름차순으로 출력
for i in range(2, N + 1):
	if prime[i] == True:
		print(i)
